apps/aecontract/test/aecontract_SUITE.erl

```diff
@@ -231,3 +245,108 @@ state_tree(_Cfg) ->
     Ct1   = ?call(get_contract, Ct1),
     <<"Code for C1">> = aect_contracts:code(Ct1),
     ok.
+
+
+%%%===================================================================
+%%% Store
+%%%===================================================================
+
+create_store(_Cfg) ->
+    state(aect_test_utils:new_state()),
+    Acc1  = ?call(new_account, 100),
+    Ct1   = ?call(insert_contract, Acc1, <<"Code for C1">>),
+    Ct1   = ?call(get_contract, Ct1),
+    #{}   = aect_contracts:state(Ct1),
+    ok.
+
+update_store(_Cfg) ->
+    state(aect_test_utils:new_state()),
+    Acc1   = ?call(new_account, 100),
+    Ct1    = ?call(insert_contract, Acc1, <<"Code for C1">>),
+    Ct1    = ?call(get_contract, Ct1),
+    Store1 = #{ <<0>> => <<42>> },
+    Ct2    = aect_contracts:set_state(Store1, Ct1),
+    Ct2    = ?call(enter_contract, Ct2),
+    Ct2    = ?call(get_contract, Ct2),
+    ok.
+
+read_store(_Cfg) ->
+    state(aect_test_utils:new_state()),
+    Acc1   = ?call(new_account, 100),
+    Ct1    = ?call(insert_contract, Acc1, <<"Code for C1">>),
+    Ct1    = ?call(get_contract, Ct1),
+    Store1 = #{ <<0>> => <<42>> },
+    Ct2    = aect_contracts:set_state(Store1, Ct1),
+    Ct2    = ?call(enter_contract, Ct2),
+    Ct2    = ?call(get_contract, Ct2),
+    Store1 = aect_contracts:state(Ct2),
+    ok.
+
+
+store_zero_value(_Cfg) ->
+    state(aect_test_utils:new_state()),
+    Acc1   = ?call(new_account, 100),
+    Ct1    = ?call(insert_contract, Acc1, <<"Code for C1">>),
+    Ct1    = ?call(get_contract, Ct1),
+    Store1 = #{ <<0>> => <<42>>
+              , <<1>> => <<0>>
+              , <<2>> => <<>> },
+    Ct2    = aect_contracts:set_state(Store1, Ct1),
+    Ct2    = ?call(enter_contract, Ct2),
+    %% Empty values are removed in state tree.
+    Ct3    = ?call(get_contract, Ct2),
+    Store2 = #{ <<0>> => <<42>>
+              , <<1>> => <<0>>},
```

Comment:
Are you supposed to be able to store 0 in the store? I see that it is not `<<>>` as the numerical value `0` would correspond to, so I see why the test succeeds, but it should really be an impossible value.

Reply:
In the PMT handling there can be the value 0, in the aevm any 0 value is striped away. Other VMs/ABIs might want to allow value 0.